File: src/content/docs/topics/cs/coding-problems/arrays-and-hashing/387-first-unique-character.ts
function assert(condition: boolean, msg: string = ''): void {
    if (!condition) throw new Error(msg || 'Assertion failed');
}

function firstUniqChar(s: string): number {
    // TODO: implement
    return -1;
}

function _runTests(): void {
    assert(firstUniqChar('leetcode') === 0);
    assert(firstUniqChar('loveleetcode') === 2);
    assert(firstUniqChar('aabb') === -1);
    assert(firstUniqChar('z') === 0);
    assert(firstUniqChar('aab') === 2);
    // perf
    const t0 = performance.now();
    firstUniqChar('a'.repeat(1_000_000) + 'b');
    console.log(`perf firstUniqChar n=1000001: ${(performance.now() - t0).toFixed(1)}ms`);
    console.log('all tests pass');
}

_runTests();

File: src/content/docs/topics/cs/coding-problems/arrays-and-hashing/387-first-unique-character.go
package main

import "fmt"

func assert(condition bool, msgs ...string) {
	if !condition {
		msg := "assertion failed"
		if len(msgs) > 0 {
			msg = msgs[0]
		}
		panic(msg)
	}
}

func firstUniqChar(s string) int {
	// TODO: implement
	return 0
}

func runTests() {
	assert(firstUniqChar("leetcode") == 0, "test 1")
	assert(firstUniqChar("loveleetcode") == 2, "test 2")
	assert(firstUniqChar("aabb") == -1, "test 3")
	assert(firstUniqChar("z") == 0, "test 4")
	assert(firstUniqChar("aab") == 2, "test 5")
	fmt.Println("all tests pass")
}

func main() { runTests() }
